Update ray-tune.md (#943)
wandb/docs: docs/guides/integrations/other/ray-tune.md

W&B integrates with [Ray](https://github.com/ray-project/ray) by offering two lightweight integrations.

> One is the `WandbLoggerCallback`, which automatically logs metrics reported to Tune to the Wandb API. The other one is the `setup_wandb()` function, which can be used with the function API. It automatically initializes the Wandb API with Tune's training information. You can just use the Wandb API like you would normally do, e.g. using `wandb.log()` to log your training process.

Reviewer: If you take my edit below, it will fix this one.

W&B integrates with [Ray](https://github.com/ray-project/ray) by offering two lightweight integrations.

> One is the `WandbLoggerCallback`, which automatically logs metrics reported to Tune to the Wandb API. The other one is the `setup_wandb()` function, which can be used with the function API. It automatically initializes the Wandb API with Tune's training information. You can just use the Wandb API like you would normally do, e.g. using `wandb.log()` to log your training process.

Reviewer: ```suggestion
- The`WandbLoggerCallback` function automatically logs metrics reported to Tune to the Wandb API.
- The `setup_wandb()` function, which can be used with the function API,  automatically initializes the Wandb API with Tune's training information. You can use the Wandb API as usual. such as by using `wandb.log()` to log your training process.
```
Not sure what "the function API" (not your text) refers to. Feel free to edit this suggestion, but I think the formatting and copyedit will be easier to read.

> One is the `WandbLoggerCallback`, which automatically logs metrics reported to Tune to the Wandb API. The other one is the `setup_wandb()` function, which can be used with the function API. It automatically initializes the Wandb API with Tune's training information. You can just use the Wandb API like you would normally do, e.g. using `wandb.log()` to log your training process.

Reviewer: Noah tells me this test is incorrect. You can ignore it. I've filed https://wandb.atlassian.net/browse/DOCS-1117 to get it fixed.

### Parameters

`project (str)` - Name of the Wandb project. Mandatory.

`api_key_file (str)` – Path to file containing the Wandb API KEY.

`api_key (str)` – Wandb API Key. Alternative to setting `api_key_file`.

`excludes (list)` – List of metrics that should be excluded from the log.

`log_config (bool)` – Boolean indicating if the config parameter of the results dict should be logged. Defaults to False.

> `upload_checkpoints (bool)` - If True, model checkpoints will be uploaded to Wandb as artifacts. Defaults to False.

Reviewer: Hmm. So I can help suggest something, what / who is the actor doing the uploading?

### Parameters

> `project (str)` - Name of the Wandb project. Mandatory.

Reviewer: ```suggestion
`project (str)`: Name of the Wandb project. Mandatory.
```

> `api_key_file (str)` – Path to file containing the Wandb API KEY.

Reviewer: ```suggestion
`api_key_file (str)`: Path to file containing the Wandb API KEY.
```

> `api_key (str)` – Wandb API Key. Alternative to setting `api_key_file`.

Reviewer: ```suggestion
`api_key (str)`: Wandb API Key. Alternative to setting `api_key_file`.
```

> `excludes (list)` – List of metrics that should be excluded from the log.

Reviewer: ```suggestion
`excludes (list)`: List of metrics to exclude from the log.
```

> `log_config (bool)` – Boolean indicating if the config parameter of the results dict should be logged. Defaults to False.

Reviewer: ```suggestion
`log_config (bool)`: Whether to log the config parameter of the results dictionary. Defaults to False.
```

> `upload_checkpoints (bool)` - If True, model checkpoints will be uploaded to Wandb as artifacts. Defaults to False.

Reviewer: ```suggestion
`upload_checkpoints (bool)`:  If True, model checkpoints are uploaded as artifacts. Defaults to False.
```

### Example

```python
from ray import tune, train
from ray.air.integrations.wandb import WandbLoggerCallback

def train_fc(config):
    for i in range(10):
        train.report({"mean_accuracy": (i + config["alpha"]) / 10})

tuner = tune.Tuner(
    train_fc,
    param_space={
        "alpha": tune.grid_search([0.1, 0.2, 0.3]),
        "beta": tune.uniform(0.5, 1.0),
    },
    run_config=train.RunConfig(
        callbacks=[
            WandbLoggerCallback(
                project="<your-project>", api_key="<your-api-key>", log_config=True
            )
        ]
    ),
)

results = tuner.fit()
```

## setup_wandb

```python
from ray.air.integrations.wandb import setup_wandb
```

> This utility function helps initialize Wandb for use with Ray Tune. For basic usage, just call `setup_wandb()` in your training function:

Reviewer: ```suggestion
This utility function helps initialize Wandb for use with Ray Tune. For basic usage, call `setup_wandb()` in your training function:
```
